packages/api-client-core/spec/mockUrqlClient.ts

```diff
@@ -134,9 +135,30 @@ const newMockFetchFn = () => {
       if (!request) {
         throw new Error("no requests started for response pushing");
       }
+      const signal = request.args[1]?.signal;
+      if (signal && signal.aborted) {
+        throw new Error("signal on request has been aborted, can't respond to a mock fetch that has been aborted");
+      }
+
       await request.resolve(response);
     });
   };
+  fn.reportAbort = async () => {
+    await act(async () => {
+      const request = requests.shift();
+      if (!request) {
+        throw new Error("no requests started for response pushing");
+      }
+      const signal = request.args[1]?.signal;
+      if (!signal) {
+        throw new Error("no signal on request, can't report an abort that has happened");
+      }
+      if (!signal.aborted) {
+        throw new Error("signal on request has not been aborted, can't report an abort that has happened");
+      }
+      await request.reject(new Error("AbortError: The user aborted a request."));
+    });
+  };
```

Comment:
Because we're mocking the browser fetch, we're responsible for actually implementing aborts if the abort signal fires. This allows us to do that.